### PR TITLE
Merge Way of Navigating Feature Branch with main

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#17642

*Description of changes:*
- Merging the way of navigating feature branch with main

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-679-fy6f29hk.web.app

File: package.json [hunk 1]
```diff
@@ -14,35 +14,39 @@
...
     "http-status-codes": "^1.4.0",
-    "launchdarkly-js-client-sdk": "^2.18.3",
-    "lodash": "^4.17.20",
+    "launchdarkly-js-client-sdk": "^2.24.2",
+    "lodash": "^4.17.21",
     "remove-accents": "^0.4.2",
     "sbc-common-components": "3.0.3",
     "style-loader": "^1.3.0",
-    "vue": "^2.6.12",
+    "vue": "2.7.10",
```
Comment (severinbeauvais) [inline package.json:36]: Let's remember to try v2.7.14.

File: src/App.vue [hunk 2]
```diff
@@ -214,6 +253,19 @@ export default class App extends Mixins(
       sessionStorage.removeItem('NR_DATA')
     }
 
+    // if there is stored legal type for an IA then incorporate it now
```
Comment (severinbeauvais) [inline src/App.vue:256]: Ditto for a firm registration right?

File: package.json (hunk 1 above)
Comment (JazzarKarim) [reply package.json:36]: If I'm not mistaken, that'll cause build time to be super fast. At least, that was the case in our UIs. Could be different here.

File: src/components/common/app-title-cols.vue [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
-  <v-row no-gutters>
-    <v-col cols="12" md="6" lg="6" class="mt-4 pl-2" :class="{'pt-10': !isMobile}">
-      <span class="h2 colour-white">Name Request
-      </span>
-    </v-col>
-    <v-col cols="12" md="6" lg="6">
-      <Stats />
-    </v-col>
-  </v-row>
+  <div>
```
Comment (severinbeauvais) [inline src/components/common/app-title-cols.vue:2]: I'll remove this unneeded div in my PR.

File: src/components/dialogs/incorporate-now-error.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <v-dialog v-model="dialog" width="45rem" persistent :attach="attach">
+    <v-card>
+      <v-card-title id="dialog-title">Unable to Incorporate Now</v-card-title>
+
+      <v-card-text id="dialog-text">
+        <!-- display message -->
+        <div class="general-error">
+          <p>Unable to incorporate now. Please cancel or try again.</p>
```
Comment (severinbeauvais) [inline src/components/dialogs/incorporate-now-error.vue:9]: 1. Didn't the UI team want to change everything to "Register now"?
2. Is this modal also used for registrations? Maybe the text should be generalized.

-> New ticket if needed, please.

Comment (JazzarKarim) [reply src/components/dialogs/incorporate-now-error.vue:9]: I was just looking at this Sev. I noticed that the `showRegisterButton` is always false currently. Hence, for SPs/GPs, we're showing the Incorporate your Business button. 

I'm not sure about the first point. I'll double check with Yui. As for the second point, yes you're right. 

Let me double check with the UX team about that.

File: src/components/new-request/search.vue [hunk 5]
```diff
@@ -357,150 +564,293 @@ export default class NewSearch extends Mixins(CommonMixin) {
...
-  set jurisdiction (jurisdiction: string) {
-    this.setJurisdiction(jurisdiction)
+  /** Retrieve text based on selected action/flow */
+  get actionNowButtonText (): string {
+    if (this.isContinuationIn) return 'Continue In Now'
+    if (this.isAmalgamation) return 'Amalgamate Now'
+    if (this.isConversion) return 'Alter Now'
+    if (this.isRestoration) return 'Restore Now'
+    if (this.isChangeName) return 'Change Name Now'
+    return 'Incorporate Now'
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:676]: Should this be "Register Now"? Need to ask UI.

File: src/plugins/index.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
+export * from './interceptors'
 export * from './authHelper'
 export * from './featureFlags'
 export * from './getConfig'
+export * from './interceptors'
```
Comment (severinbeauvais) [inline src/plugins/index.ts:5]: I'll delete the duplicate in my PR.

File: src/components/new-request/search.vue (hunk 5 above)
Comment (JazzarKarim) [reply src/components/new-request/search.vue:676]: According to Yui, there has been lots of back and forth regarding this and the comment above (Incorporate or Register). Apparently, Janis was against using different terminologies but business staff were adamant on making the distinction between the two.

I'll be asking Ethan on what the decision has fallen on.

File: src/views/landing.vue [hunk 7]
```diff
@@ -16,6 +16,12 @@
             </keep-alive>
           </transition>
         </div>
+
+        <!-- FOR DEBUGGING -->
+        <div>getRequestActionCd={{ getRequestActionCd }}</div>
+        <div>getLocation={{ getLocation }}</div>
+        <div>getEntityTypeCd={{ getEntityTypeCd }}</div>
+        <div>isMrasJurisdiction={{ isMrasJurisdiction }}</div>
```
Comment (severinbeauvais) [inline src/views/landing.vue:24]: We need to remember to remove this! But not yet.

File: package.json (hunk 1 above)
Comment (severinbeauvais) [reply package.json:36]: Yeah, there was a problem at one time, and then it all went away when ... I can't remember! But the other UIs are at 2.7.14.

**Review by severinbeauvais — APPROVED** [review]
Wow, I'm so proud of you guys! This team made so many great changes in this project. The code looks so much better than before.

None of my comments should hold back this merge. They're just things we need to ask about or look into... for future changes.

File: package.json (hunk 1 above)
Comment (JazzarKarim) [reply package.json:36]: When we upgraded to Vite 🥳 That's why I was able to upgrade them to 2.7.14.

**Comment by JazzarKarim** [comment]
> Wow, I'm so proud of you guys! This team made so many great changes in this project. The code looks so much better than before.
> 
> None of my comments should hold back this merge. They're just things we need to ask about or look into... for future changes.

This wouldn't have been possible without your guidance with everything Sev ❤️

**Review by ketaki-deodhar — APPROVED** [review]
This looks awesome! 🎉

**Comment by JazzarKarim** [comment]
@severinbeauvais OK to merge now? Shall I press on the button? 😁

**Comment by severinbeauvais** [comment]
> @severinbeauvais OK to merge now? Shall I press on the button? 😁

Do it :)

**Comment by JazzarKarim** [comment]
@severinbeauvais Are we squashing and merging to compress all the commits in one or Rebasing and merging? If I have to take a guess it's the latter but I'm just making sure.

**Comment by severinbeauvais** [comment]
I'd say don't squash, so that we can see all commits separately.

**Pull request merged**